x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/actions/connector_types/inference.ts

```diff
@@ -0,0 +1,518 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { IValidatedEvent } from '@kbn/event-log-plugin/server';
+
+import {
+  InferenceSimulator,
+  inferenceSuccessResponse,
+} from '@kbn/actions-simulators-plugin/server/inference_simulation';
+import { TaskErrorSource } from '@kbn/task-manager-plugin/common';
+import { FtrProviderContext } from '../../../../../common/ftr_provider_context';
+import { getUrlPrefix, ObjectRemover } from '../../../../../common/lib';
+import { getEventLog } from '../../../../../common/lib';
+
+const connectorTypeId = '.inference';
+const name = 'AI connector action';
+const secrets = {
+  apiKey: 'genAiApiKey',
+};
+
+const defaultConfig = { provider: 'openai' };
+
+// eslint-disable-next-line import/no-default-export
+export default function InferenceConnectorTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const objectRemover = new ObjectRemover(supertest);
+  const supertestWithoutAuth = getService('supertestWithoutAuth');
+  const configService = getService('config');
+  const retry = getService('retry');
+  const createConnector = async (apiUrl: string, spaceId?: string) => {
+    const { body } = await supertest
+      .post(`${getUrlPrefix(spaceId ?? 'default')}/api/actions/connector`)
+      .set('kbn-xsrf', 'foo')
+      .send({
+        name,
+        connector_type_id: connectorTypeId,
+        config: { ...defaultConfig, apiUrl },
+        secrets,
+      })
+      .expect(200);
+
+    objectRemover.add(spaceId ?? 'default', body.id, 'connector', 'actions');
+
+    return body.id;
+  };
+
+  describe('OpenAI', () => {
+    after(async () => {
+      await objectRemover.removeAll();
+    });
+    describe('action creation', () => {
+      const simulator = new InferenceSimulator({
+        returnError: false,
+        proxy: {
+          config: configService.get('kbnTestServer.serverArgs'),
+        },
+      });
+      const config = { ...defaultConfig, apiUrl: '' };
+
+      before(async () => {
+        config.apiUrl = await simulator.start();
+      });
+
+      after(() => {
+        simulator.close();
+      });
+
+      it('should return 200 when creating the connector without a default model', async () => {
```

Comment:
These tests should be also checking if the interference did or did not get created, right?  And also testing some error conditions, I guess like the case where the same infererence would be created, and so the pre-save hook fails?

Seems like we will also want some delete tests, checking to make sure the inference gets deleted.  Maybe one where the inference is ALREADY deleted, but not the connector.  What happens then?

Reply:
I opened a follow up ticket to add better tests coverage for this PR https://github.com/elastic/kibana/issues/196032